src/Polly.Core/ResiliencePipelineBuilderBase.cs

```diff
@@ -105,7 +108,14 @@ internal void AddPipelineComponent(Func<StrategyBuilderContext, PipelineComponen
         Guard.NotNull(factory);
         Guard.NotNull(options);
 
-        Validator(new ResilienceValidationContext(options, $"The '{TypeNameFormatter.Format(options.GetType())}' are invalid."));
+        var validationContext = new ResilienceValidationContext(
+            options,
+            $"The '{TypeNameFormatter.Format(options.GetType())}' are invalid.");
+
+        lock (ValidatorLock)
+        {
+            Validator(validationContext);
```

Comment:
As `Validator()` is also called on line 130, would it be better to move the locking into the actual validation implementation here?

https://github.com/App-vNext/Polly/blob/4f0d44a1fb7b984afe27a6b51fb63940876b1e3f/src/Polly.Core/Utils/ValidationHelper.cs#L31

```csharp
bool valid;

lock (ValidatorLock)
{
    valid = Validator.TryValidateObject(context.Instance, new ValidationContext(context.Instance), errors, true);
}

if (!valid)
{
    // ...
}
```